### ISSUE TITLE
NVDA 2016.3 on Windows 10 ver 1607 hangs opening LibreOffice

### ISSUE THREAD

**Issue by vsfoote** [nvaccess/nvda#6385]
Getting NVDA hang/recover (debug level log attached) in 2016.3 when trying to open recent LibreOffice 5.3/master builds running on Windows 10 Pro 64-bit ver 1607 builds.
[NVDA_IAccessible_handler_2016.3_hangs.log.zip](https://github.com/nvaccess/nvda/files/478365/NVDA_IAccessible_handler_2016.3_hangs.log.zip)

**Comment by feerrenrut** [comment]
We should investigate the severity of this. P1 for this investigation and then re-triage.

**Comment by michaelDCurran** [comment]
I do not see a freeze when opening LibreOffice writer on Windows 10 1607  
LibreOffice Version: 5.3.0.0.alpha1
Build ID: f4ca1573fcf445164c068c1046ab5d084e1b005f

Can you still reproduce this?
Can you provide exact steps?

**Comment by vsfoote** [comment]
HI Mic,

Attaching another debug log.  With LibreOffice open to the Writer module and a document with some dummy text pasted in--NVDA 2016.3 will scrape the UI, and with F10 to menu and F6 to navigate toolbars can get to the text of the document.   NVDA will sound the text, mouse over or "say all".

But, as I try to launch any action from a button object on one of the toolbars or from the Sidebar deck (say F11 to enter the Style panel), LibreOffice locks up and NVDA goes through recovery cycle.

The LibreOffice session remains in a wait/not-responding and eventually I have to terminate the process with Task Manager.  And when doing so, get an NVDA close chime.

So suspect there is an OLE accessible event that is getting hung up.  The NVDA log seems to suggest that something is amiss in the IA.

Here is a clip from the log file I've attached... I restarted LibreOffice via Task Manager a couple of times in the log.

<clip>
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (01:53:00):
Speaking [LangChangeCommand ('en_US'), u'Highlight Color Applies the current highlight color to the background of a text selection. If no text is selected, click the Highlight Color icon, select the text that you want to highlight, and then click the Highlight Color icon again. To change the highlight color, click the arrow next to the Highlight Color icon, and then click the color that you want.']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (01:53:00):
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (01:53:01):
Speaking [LangChangeCommand ('en_US'), u'Numbering Adds or removes numbering from the selected paragraphs.']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (01:53:01):
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (01:53:01):
Speaking [LangChangeCommand ('en_US'), u'Left Aligns the selected paragraph(s) to the left page margin.']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (01:53:01):
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (01:53:01):
Speaking [LangChangeCommand ('en_US'), u'Formatting']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (01:53:01):
Unsupported speech command: LangChangeCommand ('en_US')
DEBUGWARNING - watchdog._watcher (01:53:07):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 190, in <module>
  File "core.pyo", line 384, in main
  File "wx\_core.pyo", line 8657, in MainLoop
  File "wx\_core.pyo", line 7952, in MainLoop
  File "core.pyo", line 360, in Notify
  File "baseObject.pyo", line 123, in invalidateCaches
  File "comtypesMonkeyPatches.pyo", line 54, in newCpbDel
  File "comtypes\__init__.pyo", line 841, in __del__
  File "comtypes\__init__.pyo", line 1092, in Release

DEBUGWARNING - displayModel.DisplayModelTextInfo._get__storyFieldsAndRects (01:53:07):
AppModule does not have a binding handle
IO - speech.speak (01:53:07):
Speaking [LangChangeCommand ('en_US'), u'junkDummyText.odt - LibreOfficeDev Writer (Not Responding)']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (01:53:07):
Unsupported speech command: LangChangeCommand ('en_US')
DEBUGWARNING - displayModel.DisplayModelTextInfo._get__storyFieldsAndRects (01:53:07):
AppModule does not have a binding handle
DEBUGWARNING - watchdog._watcher (01:53:15):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 190, in <module>
  File "core.pyo", line 384, in main
  File "wx\_core.pyo", line 8657, in MainLoop
  File "wx\_core.pyo", line 7952, in MainLoop
  File "core.pyo", line 356, in Notify
  File "mouseHandler.pyo", line 145, in pumpAll
  File "mouseHandler.pyo", line 112, in executeMouseMoveEvent
  File "comtypesMonkeyPatches.pyo", line 54, in newCpbDel
  File "comtypes\__init__.pyo", line 841, in __del__
  File "comtypes\__init__.pyo", line 1092, in Release

IO - speech.speak (01:53:15):
Speaking [LangChangeCommand ('en_US'), u'Task Manager']
</clip>
[nvda_20161104_debug.zip](https://github.com/nvaccess/nvda/files/570810/nvda_20161104_debug.zip)

**Comment by michaelDCurran** [comment]
I'm still having trouble reproducing this.
Can you please provide clear steps to reproduce, e.g. Open writer, move to the standard toolbar with f6, tab to the save button, press enter.

**Comment by vsfoote** [comment]
Mic, Jamie

Still having NVDA annoyances with LibreOffice, but not clear it is still the exact issue as Toolbar buttons are for the most part stable. But now we are hanging up LibreOffice, Bugzilla [tdf#105712](https://bugs.documentfoundation.org/show_bug.cgi?id=105712)  there,  any call of the Special Character dialog locks up LO in a wait chain or directly crashes it.

Three ways to reach the Special Character dialog:

1. new Writer document with some typos, F7 Spell check dialog, select misspelling to enable Special Character dialog button, click the button --> immediate crash of LO with error dialog, NVDA continues to sound other desktop elements.
2. new Writer document, Insert -> Special Character --> immediate crash of LO --> same immediate crash of LO. NVDA continues to sound other desktop elements.
3. new Writer document, Standard toolbar Special Character button --> LO hangs but no error dialog,  NVDA continues to sound, and can screen scrape the LO UI--but no IA2 response or keyboard/mouse navigation.

This snippet from NVDA+F1 (at Debug level logging) following the Insert -> Special character and resulting error dialog.  Assume the LanguageChangeCommand are mostly benign, and it is just the oleacc.AccessibleObjectFromEvent with a bad parameter causing issues.

 =-snip-=

IO - speech.speak (15:47:43):
Speaking [LangChangeCommand ('en_US'), u'Special Character...  \np']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (15:47:43):
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (15:47:43):
Speaking [LangChangeCommand ('en_US'), u'SpecialCharacter...']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (15:47:43):
Unsupported speech command: LangChangeCommand ('en_US')
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:47:47):
oleacc.AccessibleObjectFromEvent with window 2165136, objectID -4 and childID 0: [Error -2147024809] The parameter is incorrect
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IA2Attributes (15:47:47):
IAccessibleObject.attributes COMError (-2147467259, 'Unspecified error', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IA2Attributes (15:47:47):
IAccessibleObject.attributes COMError (-2147467259, 'Unspecified error', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IA2Attributes (15:47:47):
IAccessibleObject.attributes COMError (-2147467259, 'Unspecified error', (None, None, None, 0, None))
IO - speech.speak (15:47:47):
Speaking [LangChangeCommand ('en_US'), u'LibreOffice 5.3 - Fatal Error  dialog']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (15:47:47):
Unsupported speech command: LangChangeCommand ('en_US')
IO - speech.speak (15:47:47):
Speaking [LangChangeCommand ('en_US'), u'OK  button']
DEBUGWARNING - synthDrivers.sapi5.SynthDriver.speak (15:47:47):
Unsupported speech command: LangChangeCommand ('en_US')
IO - inputCore.InputManager.executeGesture (15:47:49):
Input: kb(desktop):NVDA+f1

**Comment by feerrenrut** [comment]
I am going to close this issue since its hard to follow, and seems to talk about several different problems that we never seemed to be able to get to the bottom of. Instead I have opened "Moving the mouse into the toolbar freezes libreoffice" #7631 which seems like it possibly could have been the cause of the problems described here.

**Comment by Brian1Gaff** [comment]
I'm recently using Libra Office portable and have had some issues with 
things getting stuck, but I was not actually expecting it to be much to do 
with nvda.
 I just feel sometimes focus gets lost and sometimes parts of the office 
product do sit sulking for no apparent reason. No idea if this occurs in the 
installed version as I like the portability. Its of course a great shame 
Libra Office Writer still has no serviceable  say all, but I gather that 
awaits the Libra team to give us a way to follow the text through blank 
lines and scrolling.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
